edit-post/components/plugin-sidebar/index.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Children, cloneElement, compose } from '@wordpress/element';
+import { Slot, Fill, withFocusReturn } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { withPluginContext } from '../../api/plugin';
+import SidebarLayout from './sidebar-layout';
+
+/**
+ * Name of slot in which popover should fill.
+ *
+ * @type {String}
+ */
+const SLOT_NAME = 'PluginSidebar';
+
+class SidebarErrorBoundary extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = { hasError: false };
+	}
+
+	componentDidCatch() {
+		this.setState( { hasError: true } );
+	}
+
+	render() {
+		if ( this.state.hasError ) {
+			return <p className="plugin-sidebar-error">
+				{ __( 'An error occurred rendering the plugin sidebar.' ) }
+			</p>;
+		}
+		return this.props.children;
+	}
+}
+
+class PluginSidebar extends Component {
+	constructor( props ) {
+		super( props );
+
+		if ( typeof props.name !== 'string' ) {
```

Comment:
Do we need to include this validation in here? I don't feel like anything bad would happen if the non-string value would be provided. It would be converted to a string and it wouldn't be possible to activate such sidebar.

Reply:
It's a matter of providing accurate feedback to the developer I think?

Reply:
It feels like this kind of behavior should be enabled only in `development` env. I don't know, @mtias and @aduth - do we have some strategy in place?

Reply:
We shouldn't throw.

https://github.com/WordPress/gutenberg/pull/5430#discussion_r173636296